pulsaradmin/pkg/utils/topic_auto_creation_config.go

```diff
@@ -18,7 +18,7 @@
 package utils
 
 type TopicAutoCreationConfig struct {
-	Allow      bool      `json:"allowAutoTopicCreation"`
+	Allow      *bool     `json:"allowAutoTopicCreation"`
 	Type       TopicType `json:"topicType"`
 	Partitions *int      `json:"defaultNumPartitions"`
```

Comment:
```suggestion
	Allow      *bool     `json:"allowAutoTopicCreation,omitempty"`
	Type       TopicType `json:"topicType"`
	Partitions *int      `json:"defaultNumPartitions,omitempty"`
```

We should add the `omitempty` to optional fields. Otherwise, the value would be `null` when these fields are not set.

Reply:
great catch! updated